server/domain/cimonitor/VersionChecker.js

```diff
@@ -0,0 +1,89 @@
+const moment = require('moment');
+const semver = require('semver');
+const fileSystem = require('fs');
+const path = require('path');
+
+const StatusManager = require('../status/StatusManager');
+const StatusFactory = require('../status/StatusFactory');
+const GitHubApi = require('./GitHubApi');
+
+class VersionChecker {
+    constructor() {
+        this.lastVersionCheck = null;
+        this.statusKey = 'cimonitor-status-check';
+        this.currentVersion = this.getCurrentVersion();
+    }
+
+    getLatestVersion() {
+        return GitHubApi.call('repos/CIMonitor/CIMonitor/releases/latest').then(latestVersionInfo => {
+            return latestVersionInfo.tag_name;
+        });
+    }
+
+    getCurrentVersion() {
+        const root = path.resolve(`${__dirname}/../../../`);
```

Comment:
Hmm, not a big fan of this due to security implications... not sure how to deal with this in NodeJS either.

Reply:
What is the security implication you see?

Reply:
If somehow this file can be loaded from another directory, you can climb outside of your root, include other stuff, and possibly gain access to your server. Don't have a working example for you now but I feel going ../../../../ is never a really nice solution.

Reply:
It's a server-side application, you can't access the backend files in any way. Discussed in real life & agreed.